src/main/java/com/list/LinkedList.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * File: LinkedList.java
+ * Author: Tarun Jain
+ * Date: 1 Feb, 2016
+ * -----------------------------------------
+ * LinkedList is a generic data structure to store data items
+ * of any type in list fashion.  
+ * 
+ *  
+ */
+package com.list;
+
+/**
+ * LinkedList is a generic data structure to store elements in 
+ * list manner.
+ *
+ * @author Tarun Jain
+ * @version 1.0
+ * @param <Item>
+ * 
+ */
+
+public class LinkedList<Item> {
+
+	private Node head;
+	private int listCount;
+	
+	/**
+	 * creates a new empty 
+	 * LinkedList object.
+	 */
+	public LinkedList() {
+		head = new Node();
+		listCount = 0;
+	}
+	
+	/**
+	 * return the number of item in list
+	 * @return the size of list as integer
+	 */
+	public int size(){
+		return listCount;
+	}
+	
+	/**
+	 * check to see if list is Empty
+	 * @return true if list is Empty, false otherwise
+	 */
+	public boolean isEmpty(){
+		return size() == 0;
```

Comment:
this seems fine, but instead of calling size(), use internal variables like 'N' to compare or write `return head == null;`